src/Npgsql.NetTopologySuite/NetTopologySuiteHandler.cs

```diff
@@ -0,0 +1,348 @@
+
+        Task INpgsqlTypeHandler<LineString>.Write(LineString value, NpgsqlWriteBuffer buf, NpgsqlLengthCache lengthCache, NpgsqlParameter parameter, bool async)
+            => WriteCore(value, buf, lengthCache, parameter, async);
+
+        Task INpgsqlTypeHandler<IPolygon>.Write(IPolygon value, NpgsqlWriteBuffer buf, NpgsqlLengthCache lengthCache, NpgsqlParameter parameter, bool async)
+            => WriteCore(value, buf, lengthCache, parameter, async);
+
+        Task INpgsqlTypeHandler<Polygon>.Write(Polygon value, NpgsqlWriteBuffer buf, NpgsqlLengthCache lengthCache, NpgsqlParameter parameter, bool async)
+            => WriteCore(value, buf, lengthCache, parameter, async);
+
+        Task INpgsqlTypeHandler<IMultiPoint>.Write(IMultiPoint value, NpgsqlWriteBuffer buf, NpgsqlLengthCache lengthCache, NpgsqlParameter parameter, bool async)
+            => WriteCore(value, buf, lengthCache, parameter, async);
+
+        Task INpgsqlTypeHandler<MultiPoint>.Write(MultiPoint value, NpgsqlWriteBuffer buf, NpgsqlLengthCache lengthCache, NpgsqlParameter parameter, bool async)
+            => WriteCore(value, buf, lengthCache, parameter, async);
+
+        Task INpgsqlTypeHandler<IMultiLineString>.Write(IMultiLineString value, NpgsqlWriteBuffer buf, NpgsqlLengthCache lengthCache, NpgsqlParameter parameter, bool async)
+            => WriteCore(value, buf, lengthCache, parameter, async);
+
+        Task INpgsqlTypeHandler<MultiLineString>.Write(MultiLineString value, NpgsqlWriteBuffer buf, NpgsqlLengthCache lengthCache, NpgsqlParameter parameter, bool async)
+            => WriteCore(value, buf, lengthCache, parameter, async);
+
+        Task INpgsqlTypeHandler<IMultiPolygon>.Write(IMultiPolygon value, NpgsqlWriteBuffer buf, NpgsqlLengthCache lengthCache, NpgsqlParameter parameter, bool async)
+            => WriteCore(value, buf, lengthCache, parameter, async);
+
+        Task INpgsqlTypeHandler<MultiPolygon>.Write(MultiPolygon value, NpgsqlWriteBuffer buf, NpgsqlLengthCache lengthCache, NpgsqlParameter parameter, bool async)
+            => WriteCore(value, buf, lengthCache, parameter, async);
+
+        Task INpgsqlTypeHandler<IGeometryCollection>.Write(IGeometryCollection value, NpgsqlWriteBuffer buf, NpgsqlLengthCache lengthCache, NpgsqlParameter parameter, bool async)
+            => WriteCore(value, buf, lengthCache, parameter, async);
+
+        Task INpgsqlTypeHandler<GeometryCollection>.Write(GeometryCollection value, NpgsqlWriteBuffer buf, NpgsqlLengthCache lengthCache, NpgsqlParameter parameter, bool async)
+            => WriteCore(value, buf, lengthCache, parameter, async);
+
+        Task WriteCore(IGeometry value, NpgsqlWriteBuffer buf, NpgsqlLengthCache lengthCache, NpgsqlParameter parameter, bool async)
+        {
+            _writer.Write(value, new WriteStream(buf));
+#if NET45
+            return Task.Delay(0);
+#else
+            return Task.CompletedTask;
+#endif
+        }
+
+        sealed class WriteStream : Stream
```

Comment:
Maybe refactor this into NpgsqlWriteBuffer, a bit like `NpgsqlReadBuffer.GetStream()`?

Reply:
I like the idea, but let's do that by an additional PR since it would be good to have tests for that stream.

Reply:
As you wish, no problem.